relay-server/src/actors/project_buffer.rs

```diff
@@ -189,11 +213,130 @@ impl Service for BufferService {
     }
 }
 
-impl Drop for BufferService {
+impl Drop for MemoryBufferService {
     fn drop(&mut self) {
         let count: usize = self.buffer.values().map(|v| v.len()).sum();
         if count > 0 {
             relay_log::error!("dropped queue with {} envelopes", count);
         }
     }
 }
+
+/// [`Buffer`] interface implementation backed by SQLite.
+#[derive(Debug)]
+pub struct SqliteBufferService {
+    db: Pool<Sqlite>,
+}
+
+impl SqliteBufferService {
+    /// Creates a new [`SqliteBufferService`] from the provide path to the sqlite database file.
```

Comment:
nit + to be consistent with the naming, as per the docs above too.
```suggestion
    /// Creates a new [`SqliteBufferService`] from the provided path to the SQLite database file.
```